lib/lookdevXUsd/UsdMxVersionUpgrade.cpp

```diff
@@ -0,0 +1,1097 @@
+    _tokens->ND_swizzle_color3_color3,
+    _tokens->ND_swizzle_color3_color4,
+    _tokens->ND_swizzle_color3_vector2,
+    _tokens->ND_swizzle_color3_vector3,
+    _tokens->ND_swizzle_color3_vector4,
+    _tokens->ND_swizzle_color4_float,
+    _tokens->ND_swizzle_color4_color3,
+    _tokens->ND_swizzle_color4_color4,
+    _tokens->ND_swizzle_color4_vector2,
+    _tokens->ND_swizzle_color4_vector3,
+    _tokens->ND_swizzle_color4_vector4,
+    _tokens->ND_swizzle_vector2_float,
+    _tokens->ND_swizzle_vector2_color3,
+    _tokens->ND_swizzle_vector2_color4,
+    _tokens->ND_swizzle_vector2_vector2,
+    _tokens->ND_swizzle_vector2_vector3,
+    _tokens->ND_swizzle_vector2_vector4,
+    _tokens->ND_swizzle_vector3_float,
+    _tokens->ND_swizzle_vector3_color3,
+    _tokens->ND_swizzle_vector3_color4,
+    _tokens->ND_swizzle_vector3_vector2,
+    _tokens->ND_swizzle_vector3_vector3,
+    _tokens->ND_swizzle_vector3_vector4,
+    _tokens->ND_swizzle_vector4_float,
+    _tokens->ND_swizzle_vector4_color3,
+    _tokens->ND_swizzle_vector4_color4,
+    _tokens->ND_swizzle_vector4_vector2,
+    _tokens->ND_swizzle_vector4_vector3,
+    _tokens->ND_swizzle_vector4_vector4
+};
+
+const auto kAtanNodes = _TfTokenSet{
+    _tokens->ND_atan2_float,
+    _tokens->ND_atan2_vector2,
+    _tokens->ND_atan2_vector3,
+    _tokens->ND_atan2_vector4
+};
+
+const auto kThinFilmBSDF = _TfTokenSet{
+        _tokens->ND_dielectric_bsdf,
+        _tokens->ND_conductor_bsdf,
+        _tokens->ND_generalized_schlick_bsdf
+    };
+
+const auto kSwizzleRegex = std::regex("ND_swizzle_([^_]+)_([^_]+)");
+
+const auto kMaterialXToUsdType = std::unordered_map<TfToken, SdfValueTypeName, TfToken::HashFunctor>{
+        {_tokens->float_, SdfValueTypeNames->Float},
+        {_tokens->vector2, SdfValueTypeNames->Float2},
+        {_tokens->vector3, SdfValueTypeNames->Float3},
+        {_tokens->vector4, SdfValueTypeNames->Float4},
+        {_tokens->color3, SdfValueTypeNames->Color3f},
+        {_tokens->color4, SdfValueTypeNames->Color4f},
+        {_tokens->integer, SdfValueTypeNames->Int},
+        {_tokens->boolean, SdfValueTypeNames->Bool},
+        {_tokens->filename, SdfValueTypeNames->Asset},
+        {_tokens->string, SdfValueTypeNames->String},
+    };
+
+
+
+namespace {
+
+Ufe::Path _getMaterialPath(const Ufe::Path& materialElementPath)
+{
+    if (materialElementPath.empty()) {
+        return {};
+    }
+
+    auto sceneItem = Ufe::Hierarchy::createItem(materialElementPath);
+
+    while (sceneItem && sceneItem->nodeType() != _tokens->Material) {
+        auto hierarchy = Ufe::Hierarchy::hierarchy(sceneItem);
+        sceneItem = hierarchy->parent();
+    }
+    if (!sceneItem) {
+        return {};
+    }
+    return sceneItem->path();
+}
+
+std::optional<std::string> _isLegacyMaterial(const UsdShadeMaterial& material)
+{
+    auto mxSurfaceOutput = material.GetSurfaceOutput(_tokens->mtlx);
+    if (!mxSurfaceOutput || !mxSurfaceOutput.HasConnectedSource())
+    {
```

Comment:
OK, now i"m pretty sure: I think the clang-on comment is not working... those braces are not placed correctly.

You found a loophole!

Reply:
Loophole was installed when LookdevXUsd was moved from LookdevX:
https://github.com/Autodesk/maya-usd/blob/dev/lib/lookdevXUsd/.clang-format
I would have love a good clang-format pass on this new code, but won't happen in that folder. I might disable it temporarily to sneak in a properly formatted file after I am done fixing review comments. If I do that now it will will introduce so much whitespace noise... the clang-format diff is 1211 lines.

Reply:
OK, please don't format it: )